src/app/user/my-holdings/hooks/useIsBuilder.ts

```diff
@@ -0,0 +1,22 @@
+import { useReadBuilderRegistry } from '@/shared/hooks/contracts'
+import { useAccount } from 'wagmi'
+import { zeroAddress } from 'viem'
+
+export const useIsBuilder = () => {
+  const { address } = useAccount()
+  const { data: gauge } = useReadBuilderRegistry(
+    {
+      functionName: 'builderToGauge',
+      args: [address ?? zeroAddress],
+    },
+    { enabled: !!address },
+  )
+
+  const isUserBuilder = Boolean(gauge && gauge !== zeroAddress)
+
+  return {
+    isUserBuilder,
+    gauge,
+    address,
+  }
```

Comment:
Should we follow the same pattern followed in other parts to manage loading/error states?